src/components/Buzz/BuzzPurchase/BuzzTypeSelector.tsx

```diff
@@ -0,0 +1,127 @@
+import { Button, Group, Stack, Text } from '@mantine/core';
+import { IconBolt } from '@tabler/icons-react';
+import React from 'react';
+import { useBuzzCurrencyConfig } from '~/components/Currency/useCurrencyConfig';
+import type { PurchasableBuzzType } from '~/server/schema/buzz.schema';
+
+export interface BuzzTypeSelectorProps {
+  onSelect: (type: PurchasableBuzzType) => void;
+  onCancel?: () => void;
+  title?: string | null;
+  description?: React.ReactNode | null;
+  greenButton?: {
+    text?: string;
+    description?: React.ReactNode;
+  };
+  redButton?: {
+    text?: string;
+    description?: React.ReactNode;
+  };
+}
+
+export function BuzzTypeSelector({
+  onSelect,
+  onCancel,
+  title = 'Choose which Buzz to buy',
+  description = (
+    <>
+      We offer multiple types of Buzz for flexibility and control.
+      <br />
+      Pick the option that fits your needs—whether you want to keep things safe for work or unlock
+      the platform&apos;s full creative potential. Each type is tailored for different content and
+      payment preferences.
+    </>
+  ),
+  greenButton = {
+    text: 'Green Buzz',
+    description: (
+      <>
+        Can be purchased using <b>credit cards</b>.<br />
+        Can <b>only</b> be used to generate <b>safe for work</b> content.
+      </>
+    ),
+  },
+  redButton = {
+    text: 'Red Buzz',
+    description: (
+      <>
+        Can <b>only</b> be purchased using <b>crypto</b>.<br />
+        Can be used to generate <b>NSFW</b> content as well as anything else on the site.
+      </>
+    ),
+  },
+}: BuzzTypeSelectorProps) {
+  const { classNames: redClassNames } = useBuzzCurrencyConfig('fakered');
+  const { classNames: greenClassNames } = useBuzzCurrencyConfig('green');
+
+  return (
+    <Stack align="center" justify="center" gap="md">
+      {title && (
+        <Text fw={700} fz={28} align="center" className="mb-2 text-gray-900 dark:text-gray-100">
+          {title}
+        </Text>
+      )}
+      {description && (
+        <Text fz={16} align="center" className="mb-4 text-gray-700 dark:text-gray-300">
```

Comment:
React will complain if for some reason there's a p element inside this p element, we can avoid this with polymorphism
```suggestion
        <Text component="div" fz={16} align="center" className="mb-4 text-gray-700 dark:text-gray-300">
```